web/tests/splash.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { test, expect, Browser } from '@playwright/test';
+import * as lk from './lynxkite.ts';
+
+async function newSplash(browser: Browser) {
+  const page = await browser.newPage();
+  await page.goto('/#/');
+  await page.evaluate(() => {
+    window.sessionStorage.clear();
+    window.localStorage.clear();
+    window.localStorage.setItem('workspace-drawing-board tutorial done', 'true');
+    window.localStorage.setItem('entry-selector tutorial done', 'true');
+    window.localStorage.setItem('allow data collection', 'false');
+    // Floating elements can overlap buttons and block clicks.
+    document.styleSheets[0].insertRule('.spark-status, .user-menu { position: static !important; }');
+  });
+  await page.goto('/#/dir/');
+  page.splash = new lk.Splash(page);
+  await page.splash.expectDirectoryListed('built-ins'); // Make sure the page is loaded.
+  if (await page.splash.directory('automated-tests').isVisible()) {
+    await page.splash.deleteDirectory('automated-tests');
+  }
+  await page.splash.newDirectory('automated-tests');
+  await page.splash.expectNumWorkspaces(0);
+  await page.splash.expectNumDirectories(0);
+  return page;
+}
+
+test.describe.configure({ mode: 'serial' });
```

Comment:
The Protractor tests have an intricate test framework implemented in [`web/test/declarative.js`](https://github.com/lynxkite/lynxkite/blob/main/web/test/declarative.js). Instead of a simple independent/serial execution this allows a tree of tests, where some tests preserve the state and some tests change it.

This is made possible by the `discardAllReallyIMeanIt` endpoint, which resets LynxKite completely and provides a safe starting state. I'm not fond of this, because I often have some nice workspaces for testing stuff I'm working on, and then the tests blow it away.

My hope is that we can ditch all this. We can get to a known state by (re)creating a directory for the tests. We can write each test file to be runnable on its own. They can still share code: another test could `newSplash()` from this file and just call it. We don't need to build a framework for this.

If we put the test stuff for each test file into a separate LynxKite directory, perhaps we could one day run those in parallel. But it's fine not to worry about this for now.